crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -317,47 +312,26 @@ impl CumulocityConverter {
                 }
                 _ => {
                     // if the target entity is unregistered, try to register it first using auto-registration
-                    if self.entity_store.get(&source).is_none()
-                        && self.config.enable_auto_register
-                        && source.matches_default_topic_scheme()
-                    {
-                        let auto_registered_entities = self.try_auto_register_entity(&source)?;
-                        Ok(auto_registered_entities
-                            .into_iter()
-                            .map(|reg_msg| reg_msg.into())
-                            .collect())
-                    } else {
+                    if self.entity_cache.get(&source).is_none() {
                         // On receipt of an unregistered entity data message with custom topic scheme OR
                         // one with default topic scheme itself when auto registration disabled,
```

Comment:
Also I guess we can remove the `OR ...` sentence as this condition is removed from the mapper.
```suggestion
                        // On receipt of an unregistered entity data message with custom topic scheme
```

Reply:
Resolved by d88c054